packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -161,7 +195,9 @@ library LibProposing {
             // For a new block, the next transition ID is always 1, not 0.
             nextTransitionId: 1,
             // For unverified block, its verifiedTransitionId is always 0.
-            verifiedTransitionId: 0
+            verifiedTransitionId: 0,
+            l1StateBlockNumber: params.l1StateBlockNumber,
```

Comment:
Maybe we don't need to write l1StateBlockNumber or timestamp to the storage, as long as these two values are available in the metadata. proveBlock also has metadata as calldata input.